### PR TITLE
Bugfix: MealViewModel Mistakes

### PR DESCRIPTION
**Pull request by medwardson**
Fixing bugs within the AddMeal flow:
- Nutrients are not reflecting properly. It's because every time we copy a meal, we copy the nutritional value and then add the ingredient nutrients, which double counts. The fix was to use a `deepCopy` with an empty NutritionalInformation object, so it reflects only the ingredients in the copied meal.
- Meal Occasion not showing up even when selected. This was because we were setting `initial = Meal.default()` on this page, even though it should have a value already. Removing this fixes that.
- Tags disappearing: MealViewModel's updateMealData didn't include the tags, so it removed them every time it got called to a default empty list.
- UserId is not used at all in the AddMealFlow, so removed

**Take away / advice**: When creating a Meal, never set the nutrition straight into the nutrition fact. Add ingredients and let the constructor do it.

### CODE REVIEW

**Comment by RandomUsername1315** [comment]
- @medwardson this `Meal Occasion not showing up even when selected. This was because we were setting initial = Meal.default() on this page, even though it should have a value already. Removing this fixes that` was fixed by  #161

**Comment by medwardson** [comment]
> * @medwardson this `Meal Occasion not showing up even when selected. This was because we were setting initial = Meal.default() on this page, even though it should have a value already. Removing this fixes that` was fixed by  [Bug : meal uncomplete not allowing user to proced to next  #161](https://github.com/swent-group10/polyfit/pull/161). It also caused an issue with the name of the meal (from spoonacular) not being properly displayed.

@RandomUsername1315  that's weird, I'm up to date with main and don't see the changes. Thanks for letting me know ill check out whats wrong

**Comment by medwardson** [comment]
@RandomUsername1315 it looks like you changed it in the Nutrition screen but not the additional information one, so i just updated that

**Review by aprilslis — APPROVED** [review]
Awesome! But consider including in the description that you are doing a quick fix for userID too!

File: app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt [hunk 1]
```diff
@@ -40,28 +40,32 @@ class MealViewModel @Inject constructor(private val mealRepo: MealRepository) :
...
       mealTemp: Double = _meal.value.mealTemp,
       ingredients: MutableList<Ingredient> = _meal.value.ingredients,
-      nutritionalInformation: NutritionalInformation = _meal.value.nutritionalInformation,
       firebaseID: String = _meal.value.firebaseId,
-      createdAt: LocalDate = _meal.value.createdAt
+      createdAt: LocalDate = _meal.value.createdAt,
+      tags: MutableList<MealTag> = _meal.value.tags
   ) {
+    // When we make a new Meal, we add all the ingredient values into nutritionalInfo. If we pass
+    // existing values, then its double adding
```
Comment (RandomUsername1315) [inline app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt:48]: Good catch! This is due to the fact that the init function is called during the creation of a new Meal, which causes the nutritional values to be automatically updated according to the ingredients.

File: app/src/main/java/com/github/se/polyfit/ui/screen/AdditionalMealInfoScreen.kt [hunk 2]
```diff
@@ -32,7 +31,7 @@ fun AdditionalMealInfoScreen(
     navigateBack: () -> Unit,
     navigateForward: () -> Unit
 ) {
-  val meal by mealViewModel.meal.collectAsState(initial = Meal.default())
+  val meal by mealViewModel.meal.collectAsState()
```
[the remark this answers is not in the document]
Comment (medwardson) [reply app/src/main/java/com/github/se/polyfit/ui/screen/AdditionalMealInfoScreen.kt:34]: The change in this PR is for the Nutrition Screen, not for the AdditionalMealScreen. Same change but different files :P

**Pull request merged**